extensions/xpath/deployment/src/main/java/org/apache/camel/quarkus/component/xpath/deployment/XPathProcessor.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.xpath.deployment;
+
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+
+class XPathProcessor {
+
+    private static final String FEATURE = "camel-xpath";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    ReflectiveClassBuildItem registerForReflection() {
+        return new ReflectiveClassBuildItem(false, false, "com.sun.org.apache.xml.internal.utils.FastStringBuffer");
+    }
```

Comment:
We had this XPath thread contention issue in WF Camel some time ago: https://issues.redhat.com/browse/ENTESB-10400?focusedCommentId=13720608&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13720608 It is rather unlikely that it was fixed in Xalan. We might want to set `SystemPropertyBuildItem("org.apache.xml.dtm.DTMManager", "org.apache.xml.dtm.ref.DTMManagerDefault")` WDYT?

Reply:
Ok - I added a build step for that.